### PR TITLE
[PP][OPS-5926] validate bank account details using modern BARs endpoint.

### PR DESCRIPTION
**Pull request by pawelpanasewicz**
Signed-off-by: Pawel Panasewicz <2970040+pawelpanasewicz@users.noreply.github.com>

### CODE REVIEW

File: app/bars/BarsService.scala [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+  def validateBankDetails(sortCode: String, accountNumber: String)(implicit request: Request[_]): Future[BarsValidationResult] = {
+    val validateBankDetailsRequest = ValidateBankDetailsRequest(account = Account(
+      sortCode,
+      accountNumber
+    ))
+
+    for {
+      response <- barsConnector.validateBank(validateBankDetailsRequest)
+      result = if (response.isValid) Valid(response) else Invalid(response)
```
Comment (johnrevill) [inline app/bars/BarsService.scala]: Nb: User could enter a HMRC back account in which case BARs will return a 400 response. 
Ref: https://confluence.tools.tax.service.gov.uk/display/TEC/2020/11/06/Bank-account-reputation+will+return+a+bad-request+for+HMRC+sort+codes

File: app/bars/model/ValidateBankDetailsResponse.scala [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  val isDdSupported: Option[Boolean] = {
+    val isDdAllowed: Option[Boolean] = directDebitsDisallowed.map(_ == "no")
+    val isDdInstructionAllowed: Option[Boolean] = directDebitInstructionsDisallowed.map(_ == "no")
+
+    //returns true if there is any evidence that dd is supported
+    (isDdAllowed, isDdInstructionAllowed) match {
+      case (Some(true), Some(true))   => Some(true)
+      case (Some(true), Some(false))  => Some(true)
+      case (Some(false), Some(true))  => Some(true)
+      case (Some(false), Some(false)) => Some(false)
```
Comment (johnrevill) [inline app/bars/model/ValidateBankDetailsResponse.scala:53]: I'm sure you've already checked, but these two field raise the question: Can a DD be supported if `isDdAllowed` is false whilst `isDdInstructionAllowed` is true (and vice versa)

File: app/bars/BarsService.scala (hunk 1 above)
Comment (johnrevill) [inline app/bars/BarsService.scala]: It doesn't look like the response that gets put into `Valid.obfuscatedResponse` been obfuscated with a call to the `ValidateBankDetailsResponse.obfuscate` function

File: conf/application.conf [hunk 3]
```diff
@@ -170,6 +170,11 @@ microservice {
       port = 9730
     }
 
+    bars {
+      host = localhost
+      #9078 points to direct-debit-stubs, can't point to bars-stubs because they haven't implemented that endpoint ...
```
Comment (johnrevill) [inline conf/application.conf:175]: Nb: May need to add DIRECT_DEBIT_STUBS to the SSTTP profile

File: app/bars/BarsService.scala (hunk 1 above)
Comment (pawelpanasewicz) [reply app/bars/BarsService.scala]: Thanks, good to know. Shame that the docs doesn't mention about it. 
I'll change it.

File: app/bars/model/ValidateBankDetailsResponse.scala (hunk 2 above)
Comment (pawelpanasewicz) [reply app/bars/model/ValidateBankDetailsResponse.scala:53]: I have no idea :) But I informed Ollie, Jeremy and Jose and we decided to have such heuristic for now. I'm logging bars responses in either failure or success cases so we will stabilise it later. 
For now it should not make it any worse - currently we have only modulo check.

File: app/bars/BarsService.scala (hunk 1 above)
Comment (pawelpanasewicz) [reply app/bars/BarsService.scala]: Oh yeah, you've got me :) I'll fix it.

File: conf/application.conf (hunk 3 above)
Comment (pawelpanasewicz) [reply conf/application.conf:175]: Yes, I'll add it.

**Review by johnrevill — APPROVED** [review]

**Pull request merged**